Gmail facebook loginauthorisation

stankasvecova opened the pull request:
Hello there!
I went step by step thru google documentation and set it up according to that, however it still console.logs me failure. Ups! Review please? Thanks

stankasvecova committed:
Added googleSignIn() and facebookSignIn()

All set up as per google documentation, however it console.logs failure. 
Not sure what's wrong (apart from messed up CSS that I couldn't figure out for some reason).

stankasvecova committed:
gmail and facebook signup functions added

stankasvecova committed:
gmail and facebook SignIn functions added

MrTim commented:
@stankasvecova could you please first merge approved PR in master, then update this branch and then push it again.
Because right now it contains a lot of code which I already reviewed and this branch is quite broken because for example it doesn't contain `<script src="../js/setupFirebase.js"></script>` script.

MrTim commented:
Why this is not working:
if you look into the error message which it returns you can see following
```js
code: "auth/operation-not-supported-in-this-environment"
message: "This operation is not supported in the environment this application is running on. "location.protocol" must be http, https or chrome-extension and web storage must be enabled."
```

which mean that you can't open a popup if you have it opened without running web server.

How you can set up it:
easiest way would be to install this package https://www.npmjs.com/package/serve and after that you can run following command from the root of your project: `serve` and it tell you what to do next.
If you don't know what npm and nodejs is you can read more about it https://www.npmjs.com/ and https://nodejs.org/en/about/

stankasvecova committed:
FB: needs a public url

FB: needs a public url
Serve.json was created automatically when I was deploying the project

MrTim commented on js/serve.json:
```diff
@@ -0,0 +1,12 @@
+const handler = require('serve-handler');
```
Why this file is needed?
For me everything works without any additional json file.

MrTim reviewed (commented):
@stankasvecova  Could you maybe give a hint why you created `serve.json`?
